File: sdks/java/src/main/java/com/puxbay/resources/OrdersResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.Order;

public class OrdersResource {
    private final Puxbay client;

    public OrdersResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<Order> list(int page, int pageSize) throws PuxbayException {
        // Assuming implementation update in Client to support parameterized types
        return (PaginatedResponse<Order>) client.request("GET", "orders/?page=" + page + "&page_size=" + pageSize, null, PaginatedResponse.class);
    }

    public Order get(String orderId) throws PuxbayException {
        return client.request("GET", "orders/" + orderId + "/", null, Order.class);
    }

    public Order create(Order order) throws PuxbayException {
        return client.request("POST", "orders/", order, Order.class);
    }

    public Order cancel(String orderId) throws PuxbayException {
        client.request("POST", "orders/" + orderId + "/cancel/", null, Void.class);
        return get(orderId);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/OrdersResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.Order
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class OrdersResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1, pageSize: Int = 20): PaginatedResponse<Order> =
        client.request(HttpMethod.Get, "orders/?page=$page&page_size=$pageSize")
    
    suspend fun get(orderId: String): Order =
        client.request(HttpMethod.Get, "orders/$orderId/")
    
    suspend fun create(order: Order): Order =
        client.request(HttpMethod.Post, "orders/", order)
        
    suspend fun cancel(orderId: String): Order =
        client.request(HttpMethod.Post, "orders/$orderId/cancel/")
}
